From ff362eb063ee60c9f9143e74b99c57b96d0f5f33 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Mon, 8 Mar 2021 13:04:08 +0100
Subject: [PATCH] Adapt tests to shebangs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The test files read itself and check the first line.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 t/unit/Test2/Harness/Util/File.t       | 9 +++++----
 t/unit/Test2/Harness/Util/File/Value.t | 3 ++-
 t2/caller.t                            | 5 +++--
 3 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/t/unit/Test2/Harness/Util/File.t b/t/unit/Test2/Harness/Util/File.t
index 73a1587..1560d4b 100644
--- a/t/unit/Test2/Harness/Util/File.t
+++ b/t/unit/Test2/Harness/Util/File.t
@@ -1,3 +1,4 @@
+#!/usr/bin/perl
 use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File';
 # HARNESS-DURATION-SHORT
 
@@ -31,7 +32,7 @@ ok(dies { $two->open_file }, "Cannot open file (for reading)");
 my ($line) = split /\n/, $one->maybe_read, 2;
 like(
     $line,
-    q{use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File';},
+    q{#!/usr/bin/perl},
     "Can read file (using maybe_read)"
 );
 
@@ -44,7 +45,7 @@ is(
 ($line) = split /\n/, $one->read, 2;
 like(
     $line,
-    q{use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File';},
+    q{#!/usr/bin/perl},
     "Can read file"
 );
 
@@ -71,7 +72,7 @@ is($two->read_line, undef, "cannot read lines from missing file");
 
 is(
     $one->read_line,
-    "use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File';\n",
+    "#!/usr/bin/perl\n",
     "Got first line"
 );
 
@@ -89,7 +90,7 @@ is(
 $one->reset;
 is(
     $one->read_line,
-    "use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File';\n",
+    "#!/usr/bin/perl\n",
     "Got first line again after reset"
 );
 
diff --git a/t/unit/Test2/Harness/Util/File/Value.t b/t/unit/Test2/Harness/Util/File/Value.t
index fb0a163..fe630f1 100644
--- a/t/unit/Test2/Harness/Util/File/Value.t
+++ b/t/unit/Test2/Harness/Util/File/Value.t
@@ -1,3 +1,4 @@
+#!/usr/bin/perl
 use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File::Value';
 # HARNESS-DURATION-SHORT
 
@@ -14,7 +15,7 @@ is($val, $no_tail, "trailing newline was removed from the value");
 $val = $one->read_line;
 is(
     $val,
-    "use Test2::Bundle::Extended -target => 'Test2::Harness::Util::File::Value';",
+    "#!/usr/bin/perl",
     "got line, no newline"
 );
 
diff --git a/t2/caller.t b/t2/caller.t
index 6fd43d4..3406795 100644
--- a/t2/caller.t
+++ b/t2/caller.t
@@ -1,3 +1,4 @@
+#!/usr/bin/perl
 package FooBarBaz;
 use strict;
 use warnings;
@@ -8,7 +9,7 @@ use Test2::V0;
 is([caller(0)], [], "No caller at the flat test level");
 is(__PACKAGE__, 'FooBarBaz', "inside main package");
 like(__FILE__, qr/caller\.t$/, "__FILE__ is correct");
-is(__LINE__, 11, "Got the correct line number");
+is(__LINE__, 12, "Got the correct line number");
 is($@, '', '$@ set to empty string');
 
-done_testing;
\ No newline at end of file
+done_testing;
-- 
2.26.2

